firestore/query.go

```diff
@@ -1101,24 +1234,33 @@ type DocumentIterator struct {
 // always concrete types, and the fact that this one has two different implementations
 // is an internal detail.
 type docIterator interface {
-	next() (*DocumentSnapshot, error)
+	next(opts ...RunOption) (*DocumentSnapshot, error)
```

Comment:
Will there ever be an instance where you call the same iterator with different options, or will the options be consistent across multiple calls of `next`? I'd only be OK adding parameters to `next` if we're going to be calling the same iterator with different parameters, because this interface change might open us up to bugs in the future if that situation does arise.

It would also make unnecessary the change on line 1519.

Reply:
Design has been updated. next does not take parameters now